bff-tiltaksarrangor/src/main/kotlin/no/nav/amt/tiltak/bff/tiltaksarrangor/VeilederController.kt

```diff
@@ -0,0 +1,125 @@
+		val ansatt = hentInnloggetAnsatt()
+		arrangorAnsattTilgangService.verifiserTilgangTilGjennomforing(
+			ansatt.id,
+			request.gjennomforingId,
+			ArrangorAnsattRolle.KOORDINATOR,
+		)
+
+		val veiledere = request.veiledere.map { ArrangorVeilederInput(it.ansattId, it.erMedveileder) }
+
+		arrangorVeilederService.opprettVeiledere(veiledere, request.deltakerIder)
+	}
+
+	@ProtectedWithClaims(issuer = Issuer.TOKEN_X)
+	@GetMapping(params = ["deltakerId"])
+	fun hentVeiledereForDeltaker(@RequestParam("deltakerId") deltakerId: UUID) : List<VeilederDto> {
+		val ansatt = hentInnloggetAnsatt()
+
+		verifiserTilgangTilDeltaker(ansatt, deltakerId)
+
+		val veiledere = arrangorVeilederService.hentVeiledereForDeltaker(deltakerId)
+
+		return mapTilVeilederDtoer(veiledere)
+	}
+
+	@ProtectedWithClaims(issuer = Issuer.TOKEN_X)
+	@GetMapping(params = ["gjennomforingId"])
+	fun hentAktiveVeiledereForGjennomforing(@RequestParam("gjennomforingId") gjennomforingId: UUID) : List<VeilederDto> {
+		val ansatt = hentInnloggetAnsatt()
+
+		arrangorAnsattTilgangService.verifiserTilgangTilGjennomforing(
+			ansatt.id,
+			gjennomforingId,
+			ArrangorAnsattRolle.KOORDINATOR,
+		)
+
+		val veiledere = arrangorVeilederService.hentAktiveVeiledereForGjennomforing(gjennomforingId)
+
+		return mapTilVeilederDtoer(veiledere)
+	}
+
+	@ProtectedWithClaims(issuer = Issuer.TOKEN_X)
+	@GetMapping("/tilgjengelig")
+	fun hentTilgjengeligeVeiledere(@RequestParam("gjennomforingId") gjennomforingId: UUID) : List<TilgjengeligVeilederDto> {
+		val ansatt = hentInnloggetAnsatt()
+		arrangorAnsattTilgangService.verifiserTilgangTilGjennomforing(
+			ansatt.id,
+			gjennomforingId,
+			ArrangorAnsattRolle.KOORDINATOR,
+		)
+
+		val tilgjengelige = arrangorVeilederService.hentTilgjengeligeVeiledereForGjennomforing(gjennomforingId)
+
+		return tilgjengelige.map { TilgjengeligVeilederDto(it.id, it.fornavn, it.mellomnavn, it.etternavn) }
+	}
+
+	private fun hentInnloggetAnsatt(): Ansatt {
+		val ansattPersonligIdent = authService.hentPersonligIdentTilInnloggetBruker()
+		return arrangorAnsattService.getAnsattByPersonligIdent(ansattPersonligIdent)
+			?: throw UnauthorizedException("Arrangor ansatt finnes ikke")
+	}
+
+	private fun mapTilVeilederDtoer(veiledere: List<ArrangorVeileder>): List<VeilederDto> {
+		val ansatte = arrangorAnsattService.getAnsatte(veiledere.map { it.ansattId })
+
+		return veiledere.map { veileder ->
+			val ansatt = ansatte.find { veileder.ansattId == it.id } ?:
+			throw IllegalStateException("Fant ikke ansatt ${veileder.ansattId} for veileder ${veileder.id}")
+
+			return@map VeilederDto(
+				id = veileder.id,
+				ansattId = ansatt.id,
+				deltakerId = veileder.deltakerId,
+				erMedveileder = veileder.erMedveileder,
+				fornavn = ansatt.fornavn,
+				mellomnavn = ansatt.mellomnavn,
+				etternavn = ansatt.etternavn,
+			)
+		}
+	}
+
+	private fun verifiserTilgangTilDeltaker(ansatt: Ansatt, deltakerId: UUID) {
```

Comment:
Det begynner å bli litt spaghetti rundt tilgangskontroll for to forskjellige roller her. Denne burde kanskje være en del av `ArrangorAnsattTilgangService`, men da blir det for øyeblikket en sirkulær avhengighet mellom veileder- og tilgangsservicen.

Denne og `hentVeiledereForDeltaker` burde kanskje avventes litt med, det var vel strengt talt ikke en nødvendig del av denne oppgaven.